### PR TITLE
feature/register

### PR DESCRIPTION
**Pull request by maxence-lefebvre**

### CODE REVIEW

File: config/webpack.config.dev.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
-'use strict';
```
Comment (gautier-lefebvre) [inline config/webpack.config.dev.js]: Empty line.
We should add "maxBOF": 0 to .eslintrc on no-multiple-empty-lines.

File: src/renderer/react/app/register/Register.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+import { setKey } from 'services/state';
+
+import RegisterPresentation from './RegisterPresentation';
+
+export const RegisterContainer = Presentation => class Register extends React.PureComponent {
```
Comment (gautier-lefebvre) [inline src/renderer/react/app/register/Register.jsx]: HOCs shouldn't be Pure. It should leave the re-render logic to the presentation.

File: src/renderer/react/components/form/password/PasswordStrength.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  /**
+   * (MEMOIZED).
+   * Calculates password strength through zxcvbn.
+   *
+   * @param {string} password - The user password.
+   * @returns {{score: number}} - User password's strength.
+   * @example const { score } = this.getStrength('azerty123');
+   */
+  getStrength = memoize(password => (password.length === 0 ? { score: -1 } : zxcvbn(password)));
```
Comment (gautier-lefebvre) [inline src/renderer/react/components/form/password/PasswordStrength.jsx:29]: If you only use the score, maybe return it directly ?

File: src/renderer/react/components/general/button/Button.jsx [hunk 4]
```diff
@@ -19,7 +19,7 @@ Button.propTypes = {
   /** Boolean indicating whether the button should render as disabled */
   disabled: PropTypes.bool,
   /** Button label. */
-  label: PropTypes.string.isRequired,
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
```
Comment (gautier-lefebvre) [inline src/renderer/react/components/general/button/Button.jsx]: PropTypes.string is already included in PropTypes.node

File: src/renderer/services/css.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/**
```
Comment (gautier-lefebvre) [inline src/renderer/services/css.js]: I think we can use the `classnames` npm module, it does exactly the same thing :'(

File: src/renderer/services/state/index.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { send } from 'services/ipc';
+import { SET_KEY } from 'common/events';
+
+/**
+ * Sends the password to main process.
+ *
+ * @param {string} key - The user password.
+ * @returns {Promise<void>} - The promise to await.
+ */
+export const setKey = async key => send(SET_KEY, key);
```
Comment (gautier-lefebvre) [inline src/renderer/services/state/index.js]: You don't need to specify `async` if you don't use `await`. You can use `await` on anything.

File: config/webpack.config.dev.js (hunk 1 above)
Comment (maxence-lefebvre) [reply config/webpack.config.dev.js]: Resolved

File: src/renderer/react/app/register/Register.jsx (hunk 2 above)
Comment (maxence-lefebvre) [reply src/renderer/react/app/register/Register.jsx]: Resolved

File: src/renderer/react/components/form/password/PasswordStrength.jsx (hunk 3 above)
Comment (maxence-lefebvre) [reply src/renderer/react/components/form/password/PasswordStrength.jsx:29]: We may use other parts of the strength (hints, etc... when design is done.)

File: src/renderer/react/components/general/button/Button.jsx (hunk 4 above)
Comment (maxence-lefebvre) [reply src/renderer/react/components/general/button/Button.jsx]: Resolved

File: src/renderer/services/state/index.js (hunk 6 above)
Comment (maxence-lefebvre) [reply src/renderer/services/state/index.js]: Resolved

File: src/renderer/services/css.js (hunk 5 above)
Comment (maxence-lefebvre) [reply src/renderer/services/css.js]: Resolved

**Review by gautier-lefebvre — APPROVED** [review]

**Pull request merged**